### PR TITLE
add contains so children routes can be recognized as selected item

### PR DESCRIPTION
**Pull request by dayesouza**

### CODE REVIEW

File: packages/components/src/Tree/Tree.hooks.ts [hunk 1]
```diff
@@ -91,10 +91,12 @@ function makeDetailedItems(
 ): TreeItem[] {
 	return items.map((item) => {
 		const { children, onClick, onExpand, selected, expanded, ...rest } = item
+		const itemSelected =
```
Comment (darthtrevino) [inline packages/components/src/Tree/Tree.hooks.ts:94]: let's streamline this a bit, and rename `itemSelected` to `selected`, and include item.selected, eeg.

```js
const { children, onClick, onExpand, expanded, selected: itemSelected...rest } = item
const selected = itemSelected || item.key === selectedKey || selectedKey?.includes(`${item.key}`/)

...
```

**Review by darthtrevino — APPROVED** [review]
Can you verify that it looks fine in the stories?

File: packages/components/src/Tree/Tree.hooks.ts (hunk 1 above)
Comment (darthtrevino) [reply packages/components/src/Tree/Tree.hooks.ts:94]: Yeah I think startsWith makes sense to use as well, I could see cases where the key is in the middle of another matched route for some reason, causing some jank selection behavior.

**Comment by dayesouza** [comment]
> Can you verify that it looks fine in the stories?

@darthtrevino Yes it does, with the existing examples. Should we create one passing the / key?

**Pull request merged**